src/generate-types.ts

```diff
@@ -5,230 +5,48 @@ import * as _ from 'lodash';
-  }
-  if (queryParams.some(param => !!param.explode)) {
-    assert(queryParams.length === 1, 'only one explode: true parameter is supported');
-    const param = queryParams[0];
-    return generateTopLevelType(op, param.schema || {});
-  }
-  const jointSchema: oas.SchemaObject = {
-    type: 'object',
-    additionalProperties: false,
-    required: queryParams.filter(param => param.required).map(param => param.name),
-    properties: queryParams.reduce((memo: any, param) => {
-      memo[param.name] = param.schema;
-      return memo;
-    }, {})
-  };
-  return generateTopLevelType(op, jointSchema);
+interface ImportDefinition {
+  importAs: string;
+  importFile: string;
 }
 
-function generateParameterType(
-  type: 'path' | 'header',
-  op: string,
-  paramSchema: undefined | ReadonlyArray<oas.ParameterObject | oas.ReferenceObject>,
-  oasSchema: oas.OpenAPIObject,
-  normalize = (name: string) => name
-) {
-  const empty = generateTopLevelType(op, { type: 'void' });
-  if (!paramSchema) {
-    return empty;
-  }
-  const schema = oautil.deref(paramSchema, oasSchema);
-  const pathParams = schema
-    .map(schema => oautil.deref(schema, oasSchema))
-    .filter(schema => schema.in === type);
-  if (pathParams.length === 0) {
-    return empty;
-  }
-  const required: string[] = [];
-  pathParams.map(paramOrRef => {
-    const param = oautil.deref(paramOrRef, oasSchema);
-    if (param.required) {
-      required.push(normalize(param.name));
-    }
-  });
-  const jointSchema: oas.SchemaObject = {
-    type: 'object',
-    additionalProperties: false,
-    required,
-    properties: pathParams.reduce((memo: any, param) => {
-      memo[normalize(param.name)] = param.schema;
-      return memo;
-    }, {})
-  };
-  return generateTopLevelType(op, jointSchema);
-}
+export type Resolve = (
+  ref: string,
+  options: Options
+) =>
+  | { importAs: string; importFrom: string; name: string; generate?: () => Promise<void> }
+  | { name: string }
+  | undefined;
 
-function generateContentSchemaType(content: oas.ContentObject) {
-  const contentTypeSchemas = Object.keys(content).map(contentType =>
-    oautil.errorTag(`contentType '${contentType}'`, () => {
-      const mediaObject: oas.MediaTypeObject = content[contentType];
-      const schema: oas.SchemaObject = {
-        type: 'object',
-        properties: {
-          contentType: {
-            type: 'string',
-            enum: [contentType]
-          },
-          value: mediaObject.schema || assert.fail('missing schema')
-        },
-        required: ['contentType', 'value'],
-        additionalProperties: false
-      };
-      return schema;
-    })
-  );
-  return {
-    oneOf: contentTypeSchemas
+export interface Options {
+  forceGenerateTypes?: boolean;
+  header: string;
+  sourceFile: string;
+  externalOpenApiImports: readonly ImportDefinition[];
+  targetFile: string;
+  resolve: Resolve;
+  externalOpenApiSpecs?: (url: string) => string | undefined;
```

Comment:
This way the IDE also shouts at you for using it

```suggestion
  /** @deprecated Consider using 'resolve' instead */
  externalOpenApiSpecs?: (url: string) => string | undefined;
```